### PR TITLE
(642a) Add navigation

### PR DESCRIPTION
**Pull request by yndajas**
## Context

The designs have a navigation menu with "List of programmes" and "My cases". The latter doesn't currently have a corresponding page in our service, but the former does

This is part one of (probably) two PRs to introduce the refer start page

## Changes in this PR

- Add the navigation with a "List of programmes" link
- A few iterations in the same area of code

### Before

Courses page:

image

Course page:

image

Dashboard page (no change):

image

### After

Courses page:

image

Course page:

image

Dashboard page (no change):

image

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Comment by yndajas** [comment]
[Make a referral start page (UI)](https://trello.com/c/DeqPNrKc/642-make-a-referral-start-page-ui)

File: server/views/partials/layout.njk [hunk 1]
```diff
@@ -53,6 +54,16 @@
 
 {% block header %}
   {% include "./header.njk" %}
+
+  {{ mojPrimaryNavigation({
```
Comment (Gweaton) [inline server/views/partials/layout.njk]: Although you've said in MOJ projects this goes above the phase banner, I think we should follow the GOV.UK design system's "must" [here](https://design-system.service.gov.uk/components/phase-banner/#how-it-works) and put it below:

> Your banner must be directly under the black GOV.UK header and colour bar.

File: server/views/partials/layout.njk [hunk 2]
```diff
@@ -32,7 +32,201 @@
...
+      !function (v, y, T) {
```
Comment (Gweaton) [inline server/views/partials/layout.njk]: Let's either put this in a formatting commit, or undo this change and have it all compressed (personal preference for the latter to reduce the amount of scrolling as this is a script copied from Azure and we probably won't need to debug/tweak it (famous last words...)

File: integration_tests/pages/page.ts [hunk 3]
```diff
@@ -83,4 +83,30 @@ export default abstract class Page {
   shouldContainBackLink(href: string): void {
     cy.get('.govuk-back-link').should('have.attr', 'href', href)
   }
+
+  checkNavigation(currentPath: string, navigationEnabled: boolean): void {
```
Comment (Gweaton) [inline integration_tests/pages/page.ts:95]: Could we call this something like `shouldShow/ContainNavigation` to make the boolean arg line up a bit more clearly with the function name?

File: server/views/partials/applicationInsightsScript.njk [hunk 4]
```diff
@@ -0,0 +1,209 @@
+<script nonce="{{ cspNonce }}" type="text/javascript">
```
Comment (Gweaton) [inline server/views/partials/applicationInsightsScript.njk:1]: Ah this addresses my comment from an earlier commit - looks like a good approach to me.

File: server/views/partials/layout.njk (hunk 2 above)
Comment (Gweaton) [reply server/views/partials/layout.njk]: The issue above is resolved, but could you take this formatting out of this commit please?

**Review by Gweaton — COMMENTED** [review]
Nice work, looks good to me, just a couple of small tweaks

File: server/views/partials/layout.njk (hunk 1 above)
Comment (yndajas) [reply server/views/partials/layout.njk]: Isn't that where it is at the moment? I noticed yesterday we're using a HMPPS header variant that doesn't have the colour bar, but at the moment the navigation is directly below the black header

Comment (Gweaton) [reply server/views/partials/layout.njk]: Sorry if it wasn't clear, I was referring to the phase banner above, so black header -> phase banner -> navigation

File: integration_tests/pages/page.ts (hunk 3 above)
Comment (yndajas) [reply integration_tests/pages/page.ts:95]: My thought was that it would be weird to have a parameter asking whether to show navigation in a method called `shouldShowNavigation`, but on reflection I suppose it does make more sense in the method signature when calling it. Does `navigationEnabled` make sense as the parameter though?

File: server/views/partials/applicationInsightsScript.njk (hunk 4 above)
Comment (yndajas) [reply server/views/partials/applicationInsightsScript.njk:1]: Actually I wanted to check with you if there's a way we can confirm this is working? Do we already have tests covering that?

File: server/views/partials/layout.njk (hunk 1 above)
Comment (yndajas) [reply server/views/partials/layout.njk]: Ah yes. I'm not sure if we can do this without hacking the CSS or replacing `block content` with `block postNavigationContent` in each view though, which is similar to something you weren't keen on before 🤔

File: integration_tests/pages/page.ts (hunk 3 above)
Comment (Gweaton) [reply integration_tests/pages/page.ts:95]: How would you feel about having two functions here, one doing the `!navigationEnabled` section and one doing the other - i.e. `should` and `shouldNot` show navigation? I don't know if there's a particular benefit to having it as one function when the conditionals are kind of doing different things.

File: server/views/partials/layout.njk (hunk 1 above)
Comment (Gweaton) [reply server/views/partials/layout.njk]: Could we add a new block to the `beforeContent` block after the `govukPhaseBanner` and override it on pages that we don't want it on (which may just be the initial dashboard page at the moment)?

Comment (Gweaton) [reply server/views/partials/layout.njk]: like we do for the `backLink`

Comment (yndajas) [reply server/views/partials/layout.njk]: Yep - it just needs some CSS hacking so that the navigation is full width but I'm working on reusing the CSS from the dashboard

**Review by Gweaton — APPROVED** [review]
Thanks for tinkering with that CSS. Have tested out locally and it's looking good!
